receiver/awscontainerinsightreceiver/internal/stores/podresourcesstore.go

```diff
@@ -0,0 +1,148 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package stores // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/stores"
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+	v1 "k8s.io/kubelet/pkg/apis/podresources/v1"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/stores/kubeletutil"
+)
+
+const (
+	socketPath        = "/var/lib/kubelet/pod-resources/kubelet.sock"
+	connectionTimeout = 10 * time.Second
+	taskTimeout       = 10 * time.Second
+)
+
+var (
+	instance *PodResourcesStore
+	once     sync.Once
+)
+
+type ContainerInfo struct {
+	podName       string
+	containerName string
+	namespace     string
+}
+
+type ResourceInfo struct {
+	resourceName string
+	deviceID     string
+}
+
+type PodResourcesClientInterface interface {
+	ListPods() (*v1.ListPodResourcesResponse, error)
+}
```

Comment:
Should this go in podresourcesclient.go?

Reply:
had this just to make testing easier, this is similar to how the podstore tests things